cirq-core/cirq/sim/classical_simulator.py

```diff
@@ -117,12 +117,26 @@ def _act_on_fallback_(self, action, qubits: Sequence['cirq.Qid'], allow_decompos
 
         Raises:
             ValueError: If initial_state shape for type np.ndarray is not equal to 1.
-                        If gate is not one of X, CNOT, SWAP, CCNOT, or a measurement.
+                        If gate is not one of X, SWAP, a controlled version of X or SWAP,
+                            or a measurement.
         """
         if isinstance(self._state.basis, np.ndarray) and len(self._state.basis.shape) != 1:
             raise ValueError('initial_state shape for type np.ndarray is not equal to 1')
         gate = action.gate if isinstance(action, ops.Operation) else action
         mapped_qubits = [self.qubit_map[i] for i in qubits]
+
+        if isinstance(gate, ops.ControlledGate):
+            control_qubits = mapped_qubits[: gate.num_controls()]
+            mapped_qubits = mapped_qubits[gate.num_controls() :]
+
+            for c, v in zip(control_qubits, gate.control_values):
```

Comment:
This loop is redundant
```suggestion
```

Reply:
whoops, sorry, that was a silly oversight. fixed!